deposit_contract_proxy.sol

```diff
@@ -286,34 +294,34 @@ library BLSSignature {
     }
 
     // Return the generator of G1.
-    function P1() private pure returns (G1Point) {
+    function P1() private pure returns (G1Point memory) {
         return G1Point(
             Fp(
                 31827880280837800241567138048534752271,
-                88385725958748408079899006800036250932223001591707578097800747617502997169851,
+                88385725958748408079899006800036250932223001591707578097800747617502997169851
             ),
             Fp(
                 11568204302792691131076548377920244452,
                 114417265404584670498511149331300188430316142484413708742216858159411894806497
-            ),
+            )
         );
     }
 
-    function decodeG1Point(bytes memory encodedX, Fp Y) private pure returns (G1Point) {
+    function decodeG1Point(bytes memory encodedX, Fp memory Y) private pure returns (G1Point memory) {
         uint a = sliceToUint(encodedX, 32, 48);
         uint b = sliceToUint(encodedX, 0, 32);
-        Fp X = Fp(a, b);
+        Fp memory X = Fp(a, b);
         return G1Point(X,Y);
     }
 
-    function decodeG2Point(bytes memory encodedX, Fp2 Y) private pure returns (G2Point) {
+    function decodeG2Point(bytes memory encodedX, Fp2 memory Y) private pure returns (G2Point memory) {
```

Comment:
just out of my curiosity, can we not pass "complex" types on the stack? (for some defn of complex...)

Reply:
No, these objects will be in memory by default and not on the stack.